truss/templates/trtllm/model/model.py

```diff
@@ -122,8 +123,14 @@ async def generate():
 
         if model_input.stream:
             return generate()
-        else:
-            if self.uses_openai_api:
-                return "".join(generate())
-            else:
-                return {"text": "".join(generate())}
+
+        try:
+            response = ""
+            async for value in generate():
+                response += value
+        except HTTPException as e:
```

Comment:
What does catching and reraising the exception here do, we don't seem to be doing anything with the exception. May be add a comment explaining.